lib/engine/game/g_1824/step/buy_sell_par_exchange_shares.rb

```diff
@@ -151,6 +151,14 @@ def allow_president_change?(corporation)
             reserved.none? { |s| s.percent == 20 }
           end
 
+          def action_is_shenanigan?(_entity, _other_entity, _action, _corporation, _corp_buying)
+            return 'Exchange of Coal Minor' if action.is_a?(Action::SpecialBuy)
+            return 'Payoff of player debt' if action.is_a?(Action::PayoffPlayerDebt)
+            return 'Partial payoff of player debt' if action.is_a?(Action::PayoffPlayerDebtPartial)
+
+            super
```

Comment:
As you're repeatedly testing the class of the `action`, you could rewrite this as a `case` statement. I'm not sure if this is clearer or not.
```suggestion
            case action
            when Action::SpecialBuy
              'Exchange of Coal Minor'
            when Action::PayoffPlayerDebt
              'Payoff of player debt'
            when Action::PayoffPlayerDebtPartial
              'Partial payoff of player debt'
            else
              super
            end
```

Reply:
You could even use the compact form of the `case` statement.
```suggestion
            case action
            when Action::SpecialBuy then 'Exchange of Coal Minor'
            when Action::PayoffPlayerDebt then 'Payoff of player debt'
            when Action::PayoffPlayerDebtPartial then 'Partial payoff of player debt'
            else super
            end
```